packages/charts/react-charting/src/components/DonutChart/DonutChart.base.tsx

```diff
@@ -265,16 +266,41 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
         focusZonePropsInHoverCard={this.props.focusZonePropsForLegendsInHoverCard}
         overflowText={this.props.legendsOverflowText}
         {...this.props.legendProps}
+        canSelectMultipleLegends={this.props.legendProps?.canSelectMultipleLegends}
+        onChange={this._onLegendChange}
       />
     );
     return legends;
   }
 
+  private _onLegendChange = (
+    selectedLegends: string[],
+    event: React.MouseEvent<HTMLButtonElement>,
+    currentLegend?: ILegend,
+  ) => {
+    this.setState({ selectedLegends });
+    if (this.props.legendProps?.onChange) {
+      this.props.legendProps.onChange(selectedLegends, event, currentLegend);
+    }
+  };
+
+  private _isLegendHovered = (legend: string): boolean => {
+    return this._noLegendsHighlighted() || this._isLegendHighlighted(legend);
```

Comment:
>this._noLegendsHighlighted() [](http://example.com/codeflow?start=11&length=28)

this is redundant

Reply:
keeping this for now as we will take up the change in experience task later.